core/templates/pages/exploration-editor-page/improvements-tab/services/improvement-playthrough-modal.controller.spec.ts

```diff
@@ -0,0 +1,293 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ImprovementPlaythoughModalController.
+ */
+
+describe('Improvement Playthough Modal Controller', function() {
+  var $scope = null;
+  var $uibModalInstance = null;
+  var AlertsService = null;
+  var PlaythroughObjectFactory = null;
+  var playthrough = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module(function($provide) {
+    $provide.value('ExplorationStatesService', {
+      getState: function() {
+        return {
+          interaction: null
+        };
+      }
+    });
+  }));
+
+  describe('when playthough actions is less than max unrelated actions' +
+    ' per block', function() {
+    var playthoughActions = [{
+      actionType: 'ExplorationStart',
+      actionCustomizationArgs: {
+        state_name: {
+          value: 'state_name1'
+        }
+      },
+      schemaVersion: ''
+    }, {
+      actionType: 'ExplorationStart',
+      actionCustomizationArgs: {
+        state_name: {
+          value: 'state_name2'
+        }
+      },
+      schemaVersion: ''
+    }];
+
+    beforeEach(angular.mock.inject(function($injector, $controller) {
+      PlaythroughObjectFactory = $injector.get('PlaythroughObjectFactory');
+      AlertsService = $injector.get('AlertsService');
+
+      playthrough = PlaythroughObjectFactory.createNew(
+        0, '1', 1, 'MultipleIncorrectSubmissions', {
+          state_name: {
+            value: 'state_name1'
+          },
+          time_spent_in_exp_in_msecs: {
+            value: 200
+          }
+        }, playthoughActions
+      );
+
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+
+      var $rootScope = $injector.get('$rootScope');
+      $scope = $rootScope.$new();
+      $controller(
+        'ImprovementPlaythoughModalController', {
+          $scope: $scope,
+          $uibModalInstance: $uibModalInstance,
+          playthrough: playthrough,
+          playthroughIndex: 0
+        });
+    }));
+
+    it('should call init when controller is initialized', function() {
+      expect($scope.playthroughIndex).toBe(0);
+      expect($scope.issueIsMultipleIncorrectSubmissions).toBe(true);
+      // After running $scope.expandActionsToRender.
+      expect($scope.canExpandActions()).toBe(false);
+
+      // Run $scope.expandActionsToRender because canExpandActions is false.
+      $scope.expandActionsToRender();
+      expect($scope.canExpandActions()).toBe(false);
+    });
+
+    it('should render issue table', function() {
+      expect($scope.renderIssueTable()).toBe(
```

Comment:
Checking for the exact string will cause more harm than good, please just assert that a regex match occurs instead.

Reply:
Done.